scalapact-shared/src/main/scala/com/itv/scalapact/shared/PendingPactSettings.scala

```diff
@@ -0,0 +1,27 @@
+package com.itv.scalapact.shared
+
+import java.time.OffsetDateTime
+
+sealed abstract case class PendingPactSettings(
```

Comment:
The pattern matching feels like it's working around the fact that some of these states are illegal? (like having pending = false and wip = true). Are wip pacts a subset of pending pacts? Wonder if we could have an ADT instead to eliminate those states at the edge of the system e.g.
```scala
sealed trait PendingPactSettings
final case object Pending extends PendingPactSettings // enablePending = true, includeWipPactsSince = omitted
final case class WipPactsSince(time: OffsetDateTime) extends PendingPactSettings // enablePending = true, includeWipPactsSince = present
// reject anything else, what was the intention? 😬 
```
Then pattern match on them during serialisation. Suppose you could treat enablePending as an implementation detail of including wip pacts and parse it into `WipPactsSince` (think this is what you're doing currently with the warning log)

I may be way off on my understanding of the feature though, please educate me 😄